### PR TITLE
Create .npmignore

### PR DESCRIPTION
**Pull request by thejameskyle**

### CODE REVIEW

**Comment by gergoerdosi** [comment]
See this issue: https://github.com/hapijs/hapi/issues/1985

**Comment by thejameskyle** [comment]
Whaat? @hueniverse I think the general consensus on this has definitely changed. `node_modules` could be a tenth of the size if people didn't send down megabytes of test directories (and other files) that aren't used.

At the very least don't include images..

**Comment by hueniverse** [comment]
@thejameskyle feel free to bring it up again in hapijs/contrib.

**Comment by arb** [comment]
I don't understand why this is such an issue? How often are you installing modules that the slight overhead causes a problem?

**Comment by thejameskyle** [comment]
> How often are you installing modules that the slight overhead causes a problem?

We install everything fresh every time we deploy or run CI. This isn't a slight overhead, if you take all of the `test/` directories in all of the `node_modules` its hundreds of megabytes.

**Comment by arb** [comment]
I agree with @hueniverse, let's reopen the discussion in hapijs/contrib since this is a cross cutting issue.

File: .npmignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+images
```
Comment (arb) [inline .npmignore]: So after it's been debating a few different places, I think the "standard" is:

```
*
!lib/**
```

Check https://github.com/hapijs/good/blob/master/.npmignore as a reference.

**Comment by arb** [comment]
@thejameskyle are you still interested in landing this? Otherwise I'm going to close and open a new PR.

**Comment by arb** [comment]
Trumped by #121

**Pull request closed without merging**